### DISCUSSION TITLE
Automatic Compilation of Nuxt.js in Monorepo Structure

### DISCUSSION THREAD

**Discussion by Ltyi** [nuxt/nuxt#25933] [Questions]
My monorepo structure is as follows:

```
monorepo/packages
|-- nuxt-kit
|　| -- lib
|　|　| -- components
|　|　|　| -- TestA.vue 
|　|　|　| -- TestB.vue 
|　index.ts
|-- nuxt-app1
|　|-- nuxt.config.js
|-- nuxt-app2
|　|-- nuxt.config.js
```

in `nuxt-kit/index.ts`

```ts
import { defineNuxtModule, addComponentsDir, createResolver } from '@nuxt/kit';

export default defineNuxtModule({
  setup() {
    const resolver = createResolver(import.meta.url);

    addComponentsDir({
      path: resolver.resolve('lib/components')
    });
  }
});
```

In nuxt-app1 and nuxt-app2's `nuxtconfig.ts`

```ts
export default defineNuxtConfig({
  devtools: { enabled: true },

  typescript: {
    shim: false
  },

  modules: ['nuxt-kit']
});
```

Everything is working fine, but the issue is that when I add a new component in nuxt-kit/lib/components/, Nuxt does not reload, need to restart Nuxt.

How can I make Nuxt automatically detect changes?

**Comment by EddyDezuraud** [comment]
Did you check out the use of Layers in Nuxt3? 
I think it could be what you're looking for.

👉 [Layers documentation](https://nuxt.com/docs/getting-started/layers)

**Comment by lr-ryan** [comment]
> Did you check out the use of Layers in Nuxt3? I think it could be what you're looking for.
> 
> 👉 [Layers documentation](https://nuxt.com/docs/getting-started/layers)

This is what I want. Thanks a lot!
